hazelcast/src/main/java/com/hazelcast/wan/WanEventService.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.wan;
+
+/**
+ * The service on which the WAN event occurred.
+ */
+public enum WanEventService {
```

Comment:
instead of this enum, can't we use regular service names `MapService.SERVICE_NAME` and `ICacheService.SERVICE_NAME`

Reply:
Initially I didn't like introducing an enum but I kinda like it in the end, you can immediately see what are the possible values and both `MapService.SERVICE_NAME` and `ICacheService.SERVICE_NAME` are internal API. So I vote for keeping this as-is. WDYT?

Reply:
up to you. but i still think using service names would be better, having it means one more place to maintain, also enum is not extendible if a user wants to add wan replication for a new data-structure. And another thing, as far as i see, usage of WanEventService is to get matching service-name.

Reply:
Reverted back to service name. It's true that we currently don't allow user-defined services but we might in the future.